src/CodeGen_C.cpp

```diff
@@ -1584,12 +1587,17 @@ void CodeGen_C::compile(const LoweredFunc &f) {
         stream << "}\n";
     }
 
-    if (is_header() && f.linkage == LinkageType::ExternalPlusMetadata) {
-        // Emit the argv version
-        stream << "int " << simple_name << "_argv(void **args) HALIDE_FUNCTION_ATTRS;\n";
-
-        // And also the metadata.
-        stream << "const struct halide_filter_metadata_t *" << simple_name << "_metadata() HALIDE_FUNCTION_ATTRS;\n";
+    if (is_header()) {
+        switch (f.linkage) {
+        case LinkageType::ExternalPlusMetadata:
+            stream << "const struct halide_filter_metadata_t *" << simple_name << "_metadata() HALIDE_FUNCTION_ATTRS;\n";
+        case LinkageType::ExternalPlusArgv: // fallthrough
```

Comment:
We always compile with `-Werror,-Wimplicit-fallthrough` enabled so this will fail unless annotated with `[[clang::fallthrough]];` -- either we should add a new HALIDE_FALLTHROUGH macro to deal with this or just make the code less clever here to avoid needing to fall thru